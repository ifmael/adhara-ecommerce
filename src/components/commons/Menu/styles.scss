@import "styles/_variables.scss";

.menu-container {
  display: flex;
  justify-content: center;
  list-style-type: none;
  font-size: 1.6rem;
  line-height: 1.9rem;
  padding: 0;
  margin: 1rem 0 1.75rem;

  .menu-item {
    margin: 0 0.5rem;

    .menu-link {
      display: block;
      text-decoration: none;
      padding: 0.5rem;

      &.custom-button {
        min-width: 9rem;
      }

      &.is-active {
        color: $mainColor;
        font-weight: bold;
        border-bottom: 0.3rem solid transparent;
        border-bottom-color: $mainColor;
      }
    }
  }
}
