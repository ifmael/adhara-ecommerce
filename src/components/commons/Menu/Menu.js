import React, { Fragment } from "react";
import { array } from "prop-types";
import { NavLink } from "react-router-dom";
import "./styles.scss";

const propTypes = {
  items: array,
};

const defaultProps = {
  items: [
    {
      title: "Sign in",
      href: "/",
    },
    {
      title: "Sign up",
      href: "/",
    },
  ],
};

const Menu = ({ items }) => {
  return (
    <ul className="menu-container" data-testid="menu-container">
      {items.map((item, index) => {
        return (
          <Fragment key={index}>
            <li className="menu-item">
              <NavLink
                exact={true}
                to={item.href}
                className="menu-link"
                activeClassName="is-active"
              >
                {item.title}
              </NavLink>
            </li>
          </Fragment>
        );
      })}
    </ul>
  );
};

Menu.propTypes = propTypes;
Menu.defaultProps = defaultProps;

export default Menu;
