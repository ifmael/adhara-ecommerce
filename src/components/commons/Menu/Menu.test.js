import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import Menu from "./Menu";

const SELECTORS = {
  MENU_CONTAINER: "menu-container",
  MY_LEARNING_BUTTON: /my learning/i,
  DISCOVER_BUTTON: /discover/i,
};

const items = [
  {
    title: "Sign in",
    href: "/sign-in",
  },
  {
    title: "Sign up",
    href: "/sign-up",
  },
];

describe("Menu Test", () => {
  test("take a snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Menu items={items} />
      </MemoryRouter>
    );
    expect(asFragment(<Menu items={items} />)).toMatchSnapshot();
  });

  test("renders menu", () => {
    render(
      <MemoryRouter>
        <Menu items={items} />
      </MemoryRouter>
    );

    const menuContainer = screen.getByTestId(SELECTORS.MENU_CONTAINER);
    const firstMenuLink = screen.getByRole("link", {
      name: "Sign in",
    });
    const lastMenuLink = screen.getByRole("link", {
      name: "Sign up",
    });

    expect(menuContainer).toBeInTheDocument();
    expect(firstMenuLink).toBeInTheDocument();
    expect(firstMenuLink.getAttribute("href")).toBe("/sign-in");
    expect(lastMenuLink).toBeInTheDocument();
    expect(lastMenuLink.getAttribute("href")).toBe("/sign-up");
  });
});
