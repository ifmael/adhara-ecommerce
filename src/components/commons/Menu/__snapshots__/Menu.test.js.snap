// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Test take a snapshot 1`] = `
<DocumentFragment>
  <ul
    class="menu-container"
    data-testid="menu-container"
  >
    <li
      class="menu-item"
    >
      <a
        class="menu-link"
        href="/sign-in"
      >
        Sign in
      </a>
    </li>
    <li
      class="menu-item"
    >
      <a
        class="menu-link"
        href="/sign-up"
      >
        Sign up
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
